finetune/base.py

```diff
@@ -297,11 +309,15 @@ def _distribute_strategy(self, visible_gpus):
 
         resolved_gpus_string = ['/gpu:{}'.format(gpu) for gpu in resolved_gpus]
         if len(resolved_gpus_string) == 1:
-            distribute_strategy = OneDeviceStrategy(resolved_gpus_string[0])
+            distribute_strategy = tf.contrib.distribute.OneDeviceStrategy(resolved_gpus_string[0])
         else:
             if self.config.per_process_gpu_memory_fraction is not None:
                 warnings.warn("Setting `per_process_gpu_memory_fraction` is currently unsupported in multi-gpu environments.")
-            distribute_strategy = CentralStorageStrategy(resolved_gpus_string or None)
+            if self.config.use_mirrored_distribution:
```

Comment:
This should probably be default, run benchmarks locally and then ask someone with non heterogeneous GPU setup to repeat. Otherwise set to a distribute_strategy kwarg and use a dict lookup.